app/src/components/ArrItem.tsx

```diff
@@ -16,29 +16,32 @@ interface ArrItemType {
   price: number;
   arr: number;
   location: string;
+  paid: boolean;
 }
 
 const ArrItem: React.FC<ArrItemProps> = ({ id }) => {
   const [arr, setArr] = React.useState<ArrItemType>();
 
-  const { ArtifactApplication } = useContractContext();
+  const { ArrRegistry } = useContractContext();
 
   React.useEffect(() => {
     const loadArr = async (): Promise<void> => {
-      const arrData = await ArtifactApplication.methods.getARR(id).call();
+      const arrData = await ArrRegistry.methods.retrieve(id).call();
       const arr = {
-        from: arrData[0],
-        to: arrData[1],
-        tokenId: arrData[2],
-        price: arrData[3] / 100,
+        from: arrData.from,
+        to: arrData.to,
+        tokenId: arrData.tokenId,
+        price: arrData.price / 100,
         arr: arrData[4] / 100,
```

Comment:
Why has everything else been replace with a . rather than  a [] but this hasn't?

Reply:
That one doesn't exist anymore

Reply:
What do you mean?

Reply:
As in I removed ArrData arr from the return values, so we have to retrieve it separately.